lib/src/connector/connection.cc

```diff
@@ -343,7 +343,7 @@ class verbose_verification {
         X509_NAME_oneline(X509_get_subject_name(cert), subject_name, 256);
         X509_NAME_oneline(X509_get_issuer_name(cert), issuer_name, 256);
         bool verified = verifier_(preverified, ctx);
-        LOG_TRACE("Verifying %1%, issued by %2%. Verified: %3%",
+        LOG_DEBUG("Verifying %1%, issued by %2%. Verified: %3%",
```

Comment:
@parisiale - FWIW, I would prefer to see a different message if the verification fails.  The Clojure client has a pretty good error message - see https://gist.github.com/Iristyle/d647cb7b867b60b9d3260c5a134d4b51#file-console-services-log-L1626 for instance:

```
Caused by: java.security.cert.CertificateException: No subject alternative DNS name matching pe-201533-priv.cloudapp.net found.
```

On the Java client side there's a complete stack trace since this situation will throw an exception.

However, on the C++ client side, this ultimately results in a series of warnings that don't clearly describe the problem.  I wasted hours debugging my master / sifting through logs / config files / etc, when a clear error message here would have immediately pointed me in the right direction.  Here's a capture of the current message:

```
2016-04-05 15:41:20.103903 DEBUG puppetlabs.cpp_pcp_client.connection:325 - WebSocket TLS initialization event; about to validate the certificate
2016-04-05 15:41:20.103903 TRACE puppetlabs.cpp_pcp_client.connection:343 - Initialized SSL context to verify broker pe-201533-priv.cloudapp.net
2016-04-05 15:41:20.150780 TRACE puppetlabs.cpp_pcp_client.connection:389 - WebSocket pre-TCP initialization event
2016-04-05 15:41:20.150780 TRACE puppetlabs.cpp_pcp_client.connection:309 - Verifying /CN=Puppet CA generated on pe-201533-priv.cloudapp.net at 2016-04-01 14:42:02 +0000, issued by /CN=Puppet CA generated on pe-201533-priv.cloudapp.net at 2016-04-01 14:42:02 +0000. Verified: 1
2016-04-05 15:41:20.150780 TRACE puppetlabs.cpp_pcp_client.connection:309 - Verifying /CN=pe-201533-priv, issued by /CN=Puppet CA generated on pe-201533-priv.cloudapp.net at 2016-04-01 14:42:02 +0000. Verified: 0
2016-04-05 15:41:20.150780 TRACE puppetlabs.cpp_pcp_client.connection:394 - WebSocket post-TCP initialization event
2016-04-05 15:41:20.150780 DEBUG puppetlabs.cpp_pcp_client.connection:362 - WebSocket on fail event - time to failure 51585 us
```

My proposal is that when `verified == false`, that a `LOG_ERROR` should be generated with a message about SSL certification verification failure.  It seems to me that `verified == false` could throw like the Java client (maybe higher up in the stack?) -- but a better error message on SSL verification failure would be an improvement.